### PR TITLE
docs: addition of param and result sections

### PR DESCRIPTION
**Pull request by hachreak**
* Improves documentation. (addresses #47)

Signed-off-by: Leonardo Rossi <leonardo.r@cern.ch>

### CODE REVIEW

File: invenio_records_ui/views.py [hunk 1]
```diff
@@ -136,11 +138,23 @@ def record_view(pid_value=None, resolver=None, template=None,
     - ``pid``
     - ``record``.
 
+    Procedure followed:
+
+    #. PID and record are resolved.
+
+    #. Permission are checked.
+
+    #. view_method is called.
```
Comment (jirikuncar) [inline invenio_records_ui/views.py]:
```
#. ``view_method`` is called
```

File: invenio_records_ui/views.py [hunk 2]
```diff
@@ -136,11 +138,23 @@ def record_view(pid_value=None, resolver=None, template=None,
     - ``pid``
     - ``record``.
 
+    Procedure followed:
+
+    #. PID and record are resolved.
+
+    #. Permission are checked.
+
+    #. `view_method` is called.
```
Comment (jirikuncar) [inline invenio_records_ui/views.py]: "\`\`view_method\`\`"

File: invenio_records_ui/views.py [hunk 3]
```diff
@@ -136,11 +138,23 @@ def record_view(pid_value=None, resolver=None, template=None,
...
+
     :param pid_value: Persistent identifier value.
     :param resolver: An instance of a persistent identifier resolver. A
         persistent identifier resolver takes care of resolving persistent
         identifiers into internal objects.
     :param template: Template to render.
+    :param permission_factory: Permission factory called to check if user has
+        enough power to execute the action.
+    :param view_method: Function that is called.
+    :returns: Tuple (pid object, record object)
```
Comment (jirikuncar) [inline invenio_records_ui/views.py]: missing dot

File: invenio_records_ui/config.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+            "methods": ["GET", "POST", "PUT", "DELETE"],
+        },
+        ...
+    }
+
+:param pid_type: Persistent identifier type for endpoint. Required.
+
+:param route: URL route (must include ``<pid_value>`` pattern). Required.
+
+:param template: Template to render. Defaults to
```
Comment (jirikuncar) [inline invenio_records_ui/config.py]: @hachreak can you unify the style of default value?

```
Defaults to ``foo``.
Default: None.
(Default: ['GET']).
```

For example to:
```
Default: ``<python value>``.
```

File: invenio_records_ui/config.py [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+:param permission_factory_imp: Import path to factory that creates a
+        permission object for a given record..
+
+:param view_imp: Import path to view function. Default: None.
+
+:param record_class: Import path of record class.
+    Default: `invenio_records.api:Record`.
+
+:param methods: List of methods supported. (Default: ['GET']).
```
Comment (jirikuncar) [inline invenio_records_ui/config.py]: https://github.com/inveniosoftware/invenio-records-ui/pull/50/files#r72590087

File: invenio_records_ui/config.py [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+:param template: Template to render. Defaults to
+    ``invenio_records_ui/detail.html``.
+
+:param permission_factory_imp: Import path to factory that creates a
+        permission object for a given record..
+
+:param view_imp: Import path to view function. Default: None.
+
+:param record_class: Import path of record class.
+    Default: `invenio_records.api:Record`.
```
Comment (jirikuncar) [inline invenio_records_ui/config.py]: https://github.com/inveniosoftware/invenio-records-ui/pull/50/files#r72590087

File: invenio_records_ui/config.py [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+
+:param route: URL route (must include ``<pid_value>`` pattern). Required.
+
+:param template: Template to render. Defaults to
+    ``invenio_records_ui/detail.html``.
+
+:param permission_factory_imp: Import path to factory that creates a
+        permission object for a given record..
+
+:param view_imp: Import path to view function. Default: None.
```
Comment (jirikuncar) [inline invenio_records_ui/config.py]: https://github.com/inveniosoftware/invenio-records-ui/pull/50/files#r72590087

File: invenio_records_ui/ext.py [hunk 8]
```diff
@@ -56,12 +60,18 @@ class InvenioRecordsUI(object):
     """
 
     def __init__(self, app=None):
-        """Extension initialization."""
+        """Extension initialization.
+
+        :param app: The Flask application. (Default: None).
```
Comment (jirikuncar) [inline invenio_records_ui/ext.py]: https://github.com/inveniosoftware/invenio-records-ui/pull/50/files#r72590087

File: invenio_records_ui/config.py [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+
+:param route: URL route (must include ``<pid_value>`` pattern). Required.
+
+:param template: Template to render.
+    (Default: ``invenio_records_ui/detail.html``)
+
+:param permission_factory_imp: Import path to factory that creates a
+        permission object for a given record.
+
+:param view_imp: Import path to view function. (Default: ``None``.)
```
Comment (jirikuncar) [inline invenio_records_ui/config.py]: `.`

File: invenio_records_ui/config.py [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Flask extension for Invenio-Records-UI."""
+
+from __future__ import absolute_import, print_function
+
+
+RECORDS_UI_BASE_TEMPLATE = "invenio_records_ui/base.html"
```
Comment (lnielsen) [inline invenio_records_ui/config.py]: I don't think this is correct.

The value is set dynamically from ``BASE_TEMPLATE``:

```python
app.config.setdefault(
          "RECORDS_UI_BASE_TEMPLATE",
          app.config.get("BASE_TEMPLATE", "invenio_records_ui/base.html")
)
```

**Pull request merged**